logical/translate_response.go

```diff
@@ -35,6 +36,34 @@ func SanitizeResponse(input *Response) *HTTPResponse {
 		}
 	}
 
+	return httpResp
+}
+
+func HTTPResponseToLogicalResponse(input *HTTPResponse) *Response {
+	logicalResp := &Response{
+		Data:     input.Data,
+		warnings: input.Warnings,
+	}
+
+	if input.LeaseID != "" {
+		logicalResp.Secret = &Secret{
+			LeaseID: input.LeaseID,
+		}
+		logicalResp.Secret.Renewable = input.Renewable
```

Comment:
I am curious to know why lines 52, 53, 63 and 64 are outside of the creation constructs.